### PR TITLE
DOCSP-26452: use struct in compound ops

### PR DESCRIPTION
**Pull request by rustagir**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-golang/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-26452
Staging - https://docs-mongodbcom-staging.corp.mongodb.com/golang/docsworker-xlarge/DOCSP-26452-struct-compound/fundamentals/crud/compound-operations/

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [ ] Did you run a grammar-check?
- [x] Are all the links working?

### CODE REVIEW

**Comment by rustagir** [comment]
This PR also fixes a spelling mistake pointed out in #220

File: source/fundamentals/crud/compound-operations.txt [hunk 1]
```diff
@@ -107,30 +123,32 @@ with the following methods:
 Example
 ```````
 
-The following example matches and deletes a document where the ``type``
-is "Assam" with the ``FindOneAndDelete()`` method:
+The following example matches and deletes a document where the
+``enrollment`` field value is less than 20 using the
+``FindOneAndDelete()`` method:
```
Comment (mongoKart) [inline source/fundamentals/crud/compound-operations.txt:128]: Suggestion:

```suggestion
The following example uses the ``FindOneAndDelete()`` method 
to match and delete the first document where the``enrollment`` 
field value is less than 20:
```

File: source/fundamentals/crud/compound-operations.txt [hunk 2]
```diff
@@ -210,11 +228,11 @@ with the following methods:
 Example
 ```````
 
-The following example performs the following actions in order with the
+The following example performs the following actions in order using the
 ``FindOneAndUpdate()`` method:
```
Comment (mongoKart) [inline source/fundamentals/crud/compound-operations.txt:232]: Suggestion:

```suggestion
The following example uses the ``FindOneAndUpdate()`` method to 
perform the following actions in order:
```

File: source/fundamentals/crud/compound-operations.txt [hunk 3]
```diff
@@ -210,11 +228,11 @@ with the following methods:
...
 
-- Matches a document where the ``type`` is "Oolong"
-- Updates the matched document's ``rating`` to ``9``
+- Matches a document where the ``title`` field value includes "Modern"
```
Comment (mongoKart) [inline source/fundamentals/crud/compound-operations.txt:234]: Issue: If this is like the Find methods in other drivers, it specifically updates the first matching document, not just an arbitrary one. Probably good to make that clear.

Suggestion:

```suggestion
- Matches the first document where the ``title`` field value includes "Modern"
```

File: source/fundamentals/crud/compound-operations.txt [hunk 4]
```diff
@@ -311,34 +330,36 @@ with the following methods:
 Example
 ```````
 
-The following example performs the following actions in order with the
+The following example performs the following actions in order using the
 ``FindOneAndReplace()`` method:
```
Comment (mongoKart) [inline source/fundamentals/crud/compound-operations.txt:334]: Suggest same change here as above

File: source/fundamentals/crud/compound-operations.txt [hunk 5]
```diff
@@ -311,34 +330,36 @@ with the following methods:
...
 
-- Matches a document where the ``type`` is "English Breakfast"
-- Replaces the matched document with a new document where the ``type`` is "Ceylon" and  ``rating`` is ``6``
+- Matches a document where the ``title`` is "Representation Theory"
```
Comment (mongoKart) [inline source/fundamentals/crud/compound-operations.txt:336]: Same suggestion as above

**Review by mongoKart — CHANGES_REQUESTED** [review]
nice. a couple small suggestions

**Pull request merged**